discovery-infra/utils.py

```diff
@@ -41,18 +60,42 @@ def run_command_with_output(command):
 
 
 @retry(tries=5, delay=3, backoff=2)
-def get_service_url_with_retries(service_name, namespace):
-    return get_service_url(service_name, namespace)
+def get_service_url_with_retries(service_name, namespace, target):
+    return get_service_url(service_name, namespace, target)
 
 
-def get_service_url(service_name, namespace):
-    try:
-        log.info(f"Getting {service_name} URL in {namespace} namespace")
-        cmd = f"minikube -n {namespace} service {service_name} --url"
-        return run_command(cmd)
-    except:
-        log.error(f"Failed to get {service_name} URL in {namespace} namespace")
-        raise
+def get_service_url(service_name, namespace, target):
+    local_mode = target == 'minikube'
+    get_url = get_local_service_url if local_mode else get_remote_service_url
+    return get_url(service_name, namespace)
+
+
+@on_exception(message='Failed to get local service url')
+def get_local_service_url(service_name, namespace):
+    log.info(f'Getting {service_name} URL in {namespace} namespace')
+    return run_command(f'minikube -n {namespace} service {service_name} --url')
+
+
+@on_exception(message='Failed to get remote service url')
+def get_remote_service_url(service, namespace):
```

Comment:
not sure i understand why we need to get remote urls?

Reply:
how otherwise will you get the assisted-service url automatically?